src/test/java/com/wikia/webdriver/testcases/infoboxbuilder/InfoboxBuilderTests.java

```diff
@@ -0,0 +1,181 @@
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.oasis.pages.TemplatePage;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.PortableInfoboxPageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.special.infoboxbuilder.SpecialInfoboxBuilderPageObject;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.special.themedesigner.SpecialThemeDesignerPageObject;
+
+import org.testng.annotations.Test;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @ownshership: Content West-Wing
+ */
+public class InfoboxBuilderTests extends NewTestTemplate {
+
+  private SpecialInfoboxBuilderPageObject builder;
+  private PortableInfoboxPageObject infobox;
+  private TemplatePage templatePage;
+  private SpecialThemeDesignerPageObject themeDesigner;
+
+  void init() {
+    builder = new SpecialInfoboxBuilderPageObject(driver);
+    infobox = new PortableInfoboxPageObject(driver);
+    themeDesigner = new SpecialThemeDesignerPageObject(driver);
+  }
+
+  @Test(groups = {"InfoboxBuilderTests_001"})
+  @Execute(asUser = User.USER, onWikia = "mediawiki119")
+  public void verifyDefaultStructure() {
+    init();
+    builder.open("verifyDefaultStructure")
+        .switchToIFrame()
+        .verifyDefaultTemplateStructure();
+  }
+
+  @Test(groups = {"InfoboxBuilderTests_002"})
+  @Execute(asUser = User.USER, onWikia = "mediawiki119")
+  public void addingComponents() {
+    init();
+    builder.open("addingComponents")
+        .switchToIFrame()
+        .verifyRowAdded()
+        .verifyTitleAdded()
+        .verifyImageAdded();
+  }
+
+  @Test(groups = {"InfoboxBuilderTests_003"})
+  @Execute(asUser = User.USER, onWikia = "mediawiki119")
+  public void savingTemplate() {
+    init();
+    templatePage = builder.open("savingTemplate")
+        .switchToIFrame()
+        .addRowComponent()
+        .addImageComponent()
+        .save();
+
+    builder.verifyCreatedTemplateName("savingTemplate", templatePage);
+  }
+
+  @Test(groups = {"InfoboxBuilderTests_003"})
+  @Execute(asUser = User.USER, onWikia = "mediawiki119")
+  public void deletingComponents() {
+    init();
+    builder.open("deletingComponents")
+        .switchToIFrame()
+        /* deleting default components */
+        .verifyDeletingTitleWithIndex(0)
+        .verifyDeletingImageWithIndex(0)
+        .verifyDeletingRowWithIndex(1)
+        .verifyDeletingRowWithIndex(0)
+        /* deleting newly added components */
+        .addRowComponent()
+        .addImageComponent()
+        .addTitleComponent()
+        .verifyDeletingRowWithIndex(0)
+        .verifyDeletingImageWithIndex(0)
+        .verifyDeletingTitleWithIndex(0);
+  }
+
+  @Test(groups = {"InfoboxBuilderTests_004"})
+  @Execute(asUser = User.USER, onWikia = "mediawiki119")
+  public void customizingComponents() {
+    init();
+    builder.open("customizingComponents") //checking and unchecking -> need randomized name
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3ATrailingCommentCheck)